core/src/main/kotlin/model/properties/PropertyContainer.kt

```diff
@@ -56,4 +56,6 @@ fun <C> C.mergeExtras(left: C, right: C): C where C : Any, C : WithExtraProperti
     val newExtras = PropertyContainer((unambiguous.flatten() + replaces).associateBy { it.key })
 
     return needingFullMerge.fold(withNewExtras(newExtras)) { acc, merger -> merger(acc, left, right) }
-}
\ No newline at end of file
+}
+
+fun <C> C.mergeExtras(elems: Collection<C>): C where C : Any, C : WithExtraProperties<C> = elems.fold(this, ::mergeExtras)
```

Comment:
Sadly it is not that simple. This will lead to the case where we merge three objects - let's call them A, B, C. We merge ordinary properties of A and B, then merge this new (A+B) object's ordinary properties with ordinary properties of C, but this may lead to unexpected behavior since we extras for (A+B) are empty. I know it is quite subtle, but it may escalate, resulting in really hard to find bugs.